cirq-google/cirq_google/engine/engine_processor.py

```diff
@@ -110,8 +113,22 @@ def get_sampler(
         Returns:
             A `cirq.Sampler` instance (specifically a `engine_sampler.ProcessorSampler`
             that will send circuits to the Quantum Computing Service
-            when sampled.1
+            when sampled.
         """
+        processor = self._inner_processor()
+        # If a run_name or config_alias is not provided, initialize them
+        # to the Processor's default values.
+        if not run_name or not device_config_name:
+            logging.info(
+                "Cannot specify only one of `run_name` and `device_config_name`. "
```

Comment:
If a user tried to specify a config then they probably didn't intend to use the default. If we automatically fix it for them, then they may not notice and wind up running the experiment with a different config than they expected. They should acknowledge the issue by fixing the incorrect config, so we should probably still raise an error.

Reply:
I considered this, but thought logging that we're switching to the default values would be sufficient since there's precedent for switching to a simulator if the user is unable to connect to the `quantum_engine`. [Example](https://github.com/quantumlib/Cirq/blob/4afe4aebb36e56f0f7dac689c671454268eac0c3/cirq-google/cirq_google/engine/qcs_notebook.py#L102). 

But, I think this is a good concern so I opted to only initialize with the default value if `run_name` and `config_alias` are not specified.